liquidctl/driver/commander_pro.py

```diff
@@ -0,0 +1,514 @@
+
+        for fan in fan_channels:
+            mode = fan_modes[fan]
+            if mode == _FAN_MODE_DC or mode == _FAN_MODE_PWM:
+                buf[0] = fan
+                self._send_command(_CMD_SET_FAN_PROFILE, buf)
+
+    def set_color(self, channel, mode_str, colors, **kwargs):
+        """Set the color of each LED.
+
+        In reality the device does not have the concept of different channels
+        or modes, but this driver provides a few for convenience.  Animations
+        still require successive calls to this API.
+
+        The 'led' channel can be used to address individual LEDs, and supports
+        the 'super-fixed', 'fixed' and 'off' modes.
+
+        In 'super-fixed' mode, each color in `colors` is applied to one
+        individual LED, successively.  LEDs for which no color has been
+        specified default to off/solid black.  This is closest to how the
+        device works.
+
+        In 'fixed' mode, all LEDs are set to the first color taken from
+        `colors`.  The `off` mode is equivalent to calling this function with
+        'fixed' and a single solid black color in `colors`.
+
+        The `colors` argument should be an iterable of one or more `[red, blue,
+        green]` triples, where each red/blue/green component is a value in the
+        range 0–255.
+
+        The table bellow summarizes the available channels, modes, and their
+        associated maximum number of colors for each device family.
+
+        | Channel  | Mode        | Num colors |
+        | -------- | ----------- | ---------- |
+        | led      | off         |          0 |
+        | led      | fixed       |          1 |
+        | led      | color_shift |          2 |
+        | led      | color_pulse |          2 |
+        | led      | color_wave  |          2 |
+        | led      | visor       |          2 |
+        | led      | blink       |          2 |
+        | led      | marquee     |          1 |
+        | led      | sequential  |          1 |
+        | led      | rainbow     |          0 |
+        | led      | rainbow2    |          0 |
+
+        """
+
+        # direction is `forward` or `backward` default forward
+        # random or alternating is determined by the number of colors specified
+        # speed is `fast`, `medium` or `slow` default fast
+        #
+
+
+        # a special mode to clear the current led settings.
+        # this is usefull if the the user wants to use a led mode for multiple devices
+        if mode_str == 'clear':
+            self._data.store('saved_effects', None)
+            return
+
+        colors = list(colors)
+        expanded = colors[:3]
+        c = itertools.chain(*((r, g, b) for r, g, b in expanded))
+        colors = list(c)
+
+        direction = kwargs.get('direction', 'forward').lower()
+        speed = kwargs.get('speed', 'medium').lower()
+        start_led = kwargs.get('start_led', 1)
+        num_leds = kwargs.get('num_leds', 1);
```

Comment:
These should be exposed _directly_ as function parameters.  This makes it easier to understand which parameters are expected by a function, as well as helps inspection of the APIs using the Python REPL or pydoc.